lib/payment_processor/braintree/webhook_handler.rb

```diff
@@ -99,7 +88,15 @@ def handle_failed_subscription_charge
 
       # This method should only be called if @notification.subscription is a subscription object
       def original_action
-        @action ||= subscription.try(:action)
+        @action ||= subscription.action
+      rescue ActiveRecord::RecordNotFound
```

Comment:
Oh goodness, you're right. I'll think about how to handle `original_action` better in the context of this comment and the one above.